IHP/View/CSSFramework.hs

```diff
@@ -57,93 +61,145 @@ instance Default CSSFramework where
-                        DateTimeInput -> renderTextField "datetime-local" formField
-                        CheckboxInput -> renderCheckboxFormField formField
-                        HiddenInput -> renderTextField "hidden" formField { disableLabel = True, disableGroup = True, disableValidationResult = True }
-                        TextareaInput -> renderTextField "text" formField
-                        SelectInput {} -> renderSelectField formField
-                        FileInput -> renderTextField "file" formField
+                formGroup renderInner
                 where
+                    renderInner = case get #fieldType formField of
+                        TextInput -> styledTextFormField cssFramework "text" formField validationResult
+                        NumberInput -> styledTextFormField cssFramework "number" formField validationResult
+                        PasswordInput -> styledTextFormField cssFramework "password" formField validationResult
+                        ColorInput -> styledTextFormField cssFramework "color" formField validationResult
+                        EmailInput -> styledTextFormField cssFramework "email" formField validationResult
+                        DateInput -> styledTextFormField cssFramework "date" formField validationResult
+                        DateTimeInput -> styledTextFormField cssFramework "datetime-local" formField validationResult
+                        CheckboxInput -> styledCheckboxFormField cssFramework formField validationResult
+                        HiddenInput -> styledTextFormField cssFramework "hidden" formField { disableLabel = True, disableGroup = True, disableValidationResult = True } validationResult
+                        TextareaInput -> styledTextFormField cssFramework "text" formField validationResult
+                        SelectInput {} -> styledSelectFormField cssFramework formField validationResult
+                        FileInput -> styledTextFormField cssFramework "file" formField validationResult
+
+                    validationResult :: Blaze.Html
+                    validationResult = unless (get #disableValidationResult formField) (styledValidationResult cssFramework formField)
+
                     -- | Wraps the input inside a @<div class="form-group">...</div>@ (unless @disableGroup = True@)
                     formGroup :: Blaze.Html -> Blaze.Html
                     formGroup renderInner = case formField of
                         FormField { disableGroup = True } -> renderInner
-                        FormField { fieldInputId } -> [hsx|<div class={formGroupClass} id={"form-group-" <> fieldInputId}>{renderInner}</div>|]
+                        FormField { fieldInputId } -> styledFormGroup cssFramework fieldInputId renderInner
 
-                    inputClass :: (Text, Bool)
-                    inputClass = ((get #styledInputClass cssFramework) formField, True)
```

Comment:
I believe `(get #styledInputClass cssFramework)` is redudent here, we can call `styledInputClass` directly. Anway, moved it into the `styled-*` functions.

Reply:
No, I was wrong. When calling it directly it skips the late-binding. So I think this should also be fixed in the Pagination/ Breadcrumbs, when one `styled` is calling the other